### ISSUE TITLE
React 16.3.1 warning: Unsafe legacy lifecycles

### ISSUE THREAD

**Issue by k4mr4n** [catamphetamine/react-phone-number-input#158]
getting this warning on `"react-phone-number-input": "^1.1.12"`
```
main.js:16649 Warning: Unsafe legacy lifecycles will not be called for components using new component APIs.

PhoneNumberInput uses getDerivedStateFromProps() but also contains the following legacy lifecycles:
  componentWillMount
  componentWillReceiveProps
```

**Comment by catamphetamine** [comment]
See https://github.com/catamphetamine/react-phone-number-input/issues/127

**Comment by catamphetamine** [comment]
FYI: I released version `2.1` which now only supports React >= 16.3
